### DISCUSSION TITLE
Nova player roles

### DISCUSSION THREAD

**Discussion by astrolamb-gaming** [Thorium-Sim/thorium-nova#28] [Ideas]
A couple of definitions:
Card - a UI element, through which a user can manipulate ship systems.
Task - a job for a user to execute via a card. 
Mission - what I by default refer to game sessions as

One of the neverending problems associated with Artemis SBS is: How do you keep all players engaged, all the time? 
If you're in combat, there's few options for the comms officer. Repeatedly spamming the "Demand Surrender" button gets old pretty quickly. If you're doing a more story-focused mission, how do you keep the weapons officer happy when they can't shoot things? This is a difficult balance to achieve for both pre-scripted and GM'd missions, and these limitations have plagued Artemis players from its genesis. 

One of the goals of Nova is to allow significant customization of cards on each screen, which immediately gives us an advantage over the very static Artemis 2 (Artemis 3 is intended to have fully customizable consoles as well). But I think it's worth having a discussion about what sort of things we can add to the "basic" player screens that will ensure that all the players remain engaged. 

Empty Epsilon, as I recall, combines the roles of science and comms to resolve the issue of comms' boredom. It also has customization available for different numbers of crew, both of which help a lot. I think that we can pick and choose particular tasks and assign them to stations based on how important each task is during different phases of a mission and how busy each station is during that phase.

A couple of big questions that are imporant for determining a good overarching layout:
1. To what extent does Thorium Nova focus on combat for beginning players? Or, rephrased: How much combat does our target audience want, assuming they are absolute beginners?
2. What are the minimal tasks required for a crew to run a ship?

With these two questions in mind, a couple of initial thoughts occur to me:

Most players probably want a decent amount of combat. Admittedly my perspective on this largely comes from my experience with Artemis players, for whom combat is expected to be a huge chuck of time, since Artemis very much revolves around combat. But it is a good way (assuming it's not _difficult_ combat) to learn basic ship systems. Let's use 60% as a base percentage of time in which a beginner crew should be in combat for a given mission. To keep engaged, I think a good rule of thumb is for each person to be _busy_ approximately 80% of the time. This allows some time to settle down, de-stress, take a bio-break, what have you, while still keeping them occupied with their tasks.

With that general criterea, let's consider the minimal tasks. The big ones:
1. Piloting the ship. Someone has to control the velocity (speed and direction) of the vessel.
2. Weapons. Someone has to designate targets, and load and fire torpedoes. 

Honestly, as far as my logic takes me, that's the bare minimum, which is great for a one (yuck) or two player crew. But there could be half a dozen (or more) players helping to run the ship. In which case, we probably want one minimal task per player. 
Subsequent "minimal" tasks, not necessarily in order:
1. Power distribution
2. Communicating with other vessels
3. Scanning... whatever needs scanned

There does seem to be a pattern here, which I honestly think is pretty engrained in me from other bridge sims. I'm used to having Helm, Weapons, Engineering, Science, and Comms. But this gets into my previous comments about players getting bored at times. _If you have ideas on how to break that mold, please speak up!!!_

This in mind, it may be good to have a few different "tiers" (for lack of a better word) of tasks.
1. Minimal - bare bones, basic helm and weapons cards
2. Recommended - add more gameplay options for a full crew
3. Advanced - this would expect players to customize their own UIs to their own preferences.

When starting a new game, I envision three groups of checkboxes/selectors of some sort. The first (minimal) would be for pilot and weapons control cards. The second (recommended) would have many more, e.g. power distribution, scanning, etc. And the final section (advanced) would have whatever else we decide to add at some point.

Note that while generally people would segregate helm vs weapons cards amongst the generally applicable roles, I imagine that they could choose to have one person controlling speed, while another controls heading.

Finally, very importantly, a player should be able to add/remove cards mid-flight as desired/needed.

Thoughts on all this? Sorry for the wall of somewhat disorganized text.

**Comment by alexanderson1993** [comment]
TL;DR: Thorium Nova should focus on narrative over action/arcade gameplay, and interdependence between the stations should help maintain crew busyness without being busy for busy's sake. 

---

Whew, there's a lot to unpack here. Hopefully I can share a bit of insight and give some alternate ways to think about this. For what it's worth, all of the bridge sim people, including the space centers in Utah, have been trying to answer this question for a while.

Artemis and Empty Epsilon sit on a different end of the spectrum than Thorium (Classic and hopefully Nova). I would call Artemis and EE Action/Arcade bridge simulators, where most of the entertainment value comes from the gameplay itself. That explains your 60% battle estimate, where combat and battle is a majority of the fun in playing the game.

Thorium Classic and Nova sit on the other end of the spectrum, where most of the entertainment comes from the story and narrative. Yes, using the controls is an important part of playing the game, but the story should provide just as much, if not more, entertainment. That means, if people aren't kept busy at their stations, they can still enjoy following the storyline like a movie.

I've described Thorium Nova as "Like watching a Star Trek movie, but you're the actor." I've also compared it to Dungeons and Dragons, which has _tons_ of downtime as one of the players calculates a particularly tricky move. The hope is that players might use this extra downtime for table talk, OOC conversations, discussing points of the mission - that kind of thing. At least, that's been my experience with the Utah space center crews.

> Let's use 60% as a base percentage of time in which a beginner crew should be in combat for a given mission.

This highlights a big difference between Artemis/EE and Thorium. I asked around to some Utah space center flight directors, and they said an average 2.5 hour mission (1 hour briefing and training, 1.5 hour simulation time) has between 15%-30% combat time. There's a lot of other things happening to fill the extra time.

- Conversations with the FD (playing different NPCs)
- Asteroid field/nebula navigation (can't hit rocks or pockets of high radiation. Also, not sure how this will work in a simulated, 3D environment, but...)
- Dealing with intruders on the ship or other internal affairs.
- Managing cargo and inventory
- Repairing and upgrading the ship
- Busywork - random stuff unrelated to the storyline (the least appealing of all, should be avoided)

Another hope is that the stations in Thorium Nova will be designed such that they are all interdependent and require collaboration from the other crew members in order to accomplish tasks. This, almost by definition, creates tasks that are meaningful, since you are helping your other crew members to accomplish their jobs.

Another thing that Utah space centers try to do is write moments into the story where each of the crew members can make a difference. This is much harder to do when you don't know how many crew you'll have during a particular flight, but it's worth considering.

> There does seem to be a pattern here, which I honestly think is pretty engrained in me from other bridge sims.

Let me try to break that pattern by sharing a few of the bridge setups for some of the physical simulators out in Utah. Since they have physical bridges, their crew counts are pretty set +/- one crew member or so.

USS Voyager (School Field Trips)
The ship is similar to a Galaxy Class ship, with 25 decks and 300 crew members.
- Captain
- Ambassador
- 1st Officer
- Engineer
- Right Wing Power
- Right Wing - Navigation/Operations (this station controls transporters, thrusters, and environment controls)
- Sensors
- Long Range Communications
- Short Range Communications
- Damage Control 1
- Damage Control 2
- Left Wing Power
- Left Wing - Engine Control
- Tactical
- Science
- Security 1
- Security 2
- Medical

That's 18 bridge crew max, which is necessary for fitting half of a typical elementary school class into the simulator while the other half is in the planetarium. Granted, this is an extreme example. Many of the kids will be sitting around for much of the flight, and the Flight Director mixes things up by having actors go to the bridge as intruders or NPCs. 

USS Odyssey
The ship is similar to a very large runabout, with 2 decks and 10 crew (2 NPCs)
- Captain
- 1st Officer
- Flight Control
- Communications
- Tactical
- Sensors
- Operations
- Engineer

8 Stations, but with no extra NPC crew members, there's a lot less stuff to do. Bridge crew very frequently doesn't have things to do, which makes the storyline much more important.

USS Phoenix
The ship is similar to the USS Defiant, with 4 decks and 50 crew.
- Captain
- 1st Officer/Sensors
- Tactical/Security
- Engineer/Damage Control
- Operations/Communications
- Flight Control/Medical

Six stations, totally stacked with tons of stuff to do because of the extra NPC crew and such a small bridge crew.

Finally, take a look at this list of all the different cards that Thorium Classic provided. I'm not saying having lots and lots of cards is a good thing (to be honest, there's about a dozen of them I wish I never made), but these are what the Utah space centers are using in all of their ships and it's working out okay so far.

My point is that there are lots of options here for how to organize the stations, especially when you are also simulating interactions between NPC crew members elsewhere on the ship.

Sorry for the totally massive response, but you brought up a lot of good points. Let me know your thoughts about these, or if there is anything I didn't address.

**Reply by alexanderson1993** [reply]
And it should go without saying, but I loved reading your thoughts. Keep them coming!

**Reply by astrolamb-gaming** [reply]
Bit of a sidenote, but theoretically we could use something like OpenAI to assist with generating random quests/missions and such. I do have access to the beta.

**Reply by alexanderson1993** [reply]
That deserves a discussion of its own. I'm particularly interested in how it could help create those side quests and missions. There's a lot of opportunity for that, both in stuff external to the ship (long range messages, interceptions, records logs randomly found in space) and internal (coming up with events on the ship for security/medical officers to deal with, generating damage reports). Those are just a few ideas, and OpenAI might be very well or poorly suited to them. In either case, we should talk more of this in the future.

**Comment by Xansta** [comment]
This discussion and the Target Demographic seem very related. I like being able to cater to a large group of people. Alex's three space center examples (18, 8 and 6 people, if I'm reading that correctly) is a nice set of different sized groups. The largest group consists primarily of elementary school age crew. You even mentioned that the class was divided: there were other kids waiting to take their turn in the simulator. You talked about strategies to keep all 18 entertained using strategies involving NPCs which I assumed were fellow volunteers to help make the event that much more memorable.

I help game master two different groups once a week. One group uses Artemis, the other uses Empty Epsilon. The ages are generally older than elementary age and we're constrained to working over the internet. Because the engines focus primarily on combat action, the GM focus is on making missions that consist largely of combat actions. Let's face it, though, both of these groups like that emphasis and we get negative feedback when missions stray too far from the combat motif.

There are likely some adults that would favor missions that were not majority combat. I know that role playing is a big thing for members of both of the groups. I can see where role playing might also be a draw for the younger participants. I can also see where the combat portion of a mission might deter some parents from letting their kids participate. Also, the kids themselves will likely get far more twitch game satisfaction from the plethora of other video games available. 

What a bridge simulator offers above normal video games and above typical space program education for kids is the sense of working together to overcome adversity, even if the adversity is simulated. 

I'm rambling. Back to my intended original topic: In both of the groups I frequently GM, groups larger than what typically fits on a ship (4-6 people) are divided into multiple ships. It seems that the typical default for flights/missions/scenarios in Nova revolves around a single ship. For a larger group of older than elementary age participants, I think the novelty of the simulation might wear off quickly if they're in largely a passive role, where their station/card/functional responsibility comes up rarely. Even in a D&D Party of 18, you're most likely going to divide that group into smaller groups to be sure all the players have a greater chance of actively participating.

I want to be sure we can handle a range of sizes, ages, capabilities and commitments. Yes, the engine will need to be able to handle that kind of range and I think the vision absolutely encompasses that, but I want to be sure we have missions/flights/scenarios that either inherently support such a diverse array or participants or are labeled clearly as designed for a particular size, age, etc.

We may also want to come up with terms for the various player roles to be used across missions. For example, in my Artemis group, they use the term Tactical and Weapons interchangeably: the person that controls the weapons on the ship. On Empty Epsilon, they are two different things. The weapons officer is responsible for targeting and shooting, but the tactical officer is responsible for flying the ship, targeting and shooting. It's quite possible that we can decouple an officer's title from their responsibilities and have the responsibilities depend on the number and type of crew that show up. Reminder: EE has something called Operations which is essentially Relay and Science together.

Simple examples with possible solutions omitting experience level (N = Thorium Nova, E = Empty Epsilon, A = Artemis):

- 3 participants
  - E: 1:Operations/Captain, 2: Tactical, 3:Engineering+/GM (or no GM)
  - A: 1:Captain/Science/Communications, 2:Helm, 3:Weapons (no GM)
  - N: 1:Captain/Flight Ops/Medical, 2:Tactical/Security/Sensors, 3:Engineering/Operations/Communications (no GM)
- 4 participants
  - A: 1:Captain/Science/Communications, 2:Helm, 3:Weapons, 4:Engineering (no GM)
  - N: 1:Captain/Medical/GM (or no GM), 2:Tactical/Security, 3:Engineering/Operations, 4:Sensors/Communications
  - E: 1:Captain/GM (or no GM), 2:Operations, 3:Tactical, 4:Engineering+

Instead of enumerating all the possible combinations, suffice it to say that roles and responsibilities can vary drastically depending on how many you have participating. Having a large number of potential roles (Helm, Weapons, Engineering, Science, Communications [Long Range, Short Range and without titular qualifications], Relay, Fighter Pilot, Operations, Tactical, Engineering+, Damage Control, First Officer, Flight Control, Security, Medical, Sensors, Power Management, Ship's Log, Ambassador, Engine Control (left and right), Captain, Commanding Officer, Executive Officer, Observer, Fleet Captain, etc.) can make it difficult to create a simulated ship and any related missions much less assign participants to the possible roles when they show up to play.

Creating stations just for make-work should be avoided, the exception being for younger or inexperienced crew that either volunteer for less responsibility or action or the teacher/parent/responsible adult can clearly see that a younger participant needs to be placed in a role with less responsibility.

From what I can tell, much of the work on Thorium was developed for one-off situations where the simulator was strongly associated to a physical set, the paradigm assumed a single ship to accommodate all participants and there was likely one mission (or flight) or perhaps a handful written for this combination. I don't want to get locked into the same roles that are typical of Artemis/Empty Epsilon, but the cards may be too flexible. I think we should seriously consider some kind of relatively standard set of functions on a station (groups of cards) that allows differing degrees of control of the simulation. This will allow player roles with different titles. The player roles based on title should have similar responsibilities between simulations and/or flights, but they don't have to be identical. There should be some kind of tiered system of titles indicating when a player has more responsibility.

This might be clearer with some kind of diagram :-)

**Reply by alexanderson1993** [reply]
Let me know if I'm totally off-base here. 

> From what I can tell, much of the work on Thorium was developed for one-off situations where the simulator was strongly associated to a physical set.

This is correct, and when I read this the lightbulb went off in my mind. This is why I appreciate having such a diverse group of contributors - it helps me recognize that notions I have for the way I think bridge simulators should be made isn't necessarily the best for Thorium Nova. I've already said that I don't intend for the Utah space centers to be the primary audience for these controls, so building according to requirements that only match their use-case is silly.

> the cards may be too flexible.

I agree with this, but I also really like the notion of cards as a means to distribute controls across many stations without having to code separate views for every station complement - by which I mean the combination of stations for a particular number of crew.

Another wrinkle - two crews with the same number of people might have totally different station complements because the ship has different capabilities. For example, the 6 person Defiant ship with 4 decks vs the 8 person Runabout with 2 decks. The former has extra controls for security, medical, and damage teams. Using cards makes it much easier to assign the necessary controls to stations based on the circumstances of the flight.

This also simplifies development a little bit. Cards are the basic unit for UI, which means developers can develop cards in isolation, without having to mess around with code for other aspects of the simulator.

A proposal: Use cards for crew controls, but have the roles and station complements for different crew sizes statically defined in code, with no ability for users to configure them. These configurations would be lists of cards assigned based on the size of the crew and the capabilities of the ship, and could likely be auto-generated by Thorium itself. Larger (7+) crews can either choose to split and run two ships for a mission, or use a ship with more capabilities and one of the larger station complements, but that's up to them.

### An Alternative To Cards

If I understand correctly, EE has different UI components for controlling various aspects of the ship, and each station composes those components together. So you have the engine speed, the sensor grid, the E-thrusters, the phasers, the scanners, the energy readings, the power controls - all of that can be split between all 6 stations, or combined onto one or two. 

Thorium Nova could easily do the same, and instead of defining station complements as combinations of cards, they are composed out of components. 

I don't really like this thought, though, since it almost encourages single-screen stations, and I think Thorium Nova works better with multi-screen stations. Could be my bias - I could be convinced otherwise. But thinking of things like a tactical station which is in charge of weapons, shields, internal sensors, and security - that's a lot to jam into a single screen. 😅

**Reply by astrolamb-gaming** [reply]
I do like how Thorium Classic has "tabs", for lack of a better word, for each station. I think that defining non-customizible tabs which can be distributed as needed would be a good compromise.

**Reply by astrolamb-gaming** [reply]
I also think that, unlike artemis and EE, thorium should encourage larger crews on fewer ships to enable the level of ship control integration that was done in Utah, even if they're online instead of in person. Personally, so long as there are sufficient people, i don't see online vs in person as a detriment. I really like this take on storytelling that doesn't revolve around combat, though i admit i don't fully understand how it will work.

**Comment by alexanderson1993** [comment]
> I do like how Thorium Classic has "tabs", for lack of a better word

I use the word "Cards" for those tabs, just to make sure we're talking about the same thing. Glad you like them - I like them too.

> I really like this take on storytelling that doesn't revolve around combat, though i admit i don't fully understand how it will work.

I don't know how it will work either, to be honest. Both you and @Xansta are absolutely right. I think the goal, then, should be to make it possible to write missions that are combat heavy and combat light, and see how test crews take to either approach.

I think a big _big_ differentiator between whether a combat light mission will work is having a flight director, since they are able to entertain the crews with antics and help the crew through particularly complicated puzzles or story points. We shall see, though. It's worthy of experimentation.
